src/button-group/__tests__/button-group-focus.test.tsx

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```

Comment:
Focus tests validate the focus function works and the focus lands on the menu trigger after the menu is closed. The unit tests do not allow for some more complex checks such as focus delegation after an item is removed or becomes disabled - these transitions are covered with the integ tests.